openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -1,14 +1,49 @@
 from rest_framework import serializers
+from django.conf import settings
 from django.contrib.auth.models import User
 from openedx.core.djangoapps.user_api.accounts import NAME_MIN_LENGTH
 from openedx.core.djangoapps.user_api.serializers import ReadOnlyFieldsSerializerMixin
 
-from student.models import UserProfile
+from student.models import UserProfile, LanguageProficiency
 from .helpers import get_profile_image_url_for_user, PROFILE_IMAGE_SIZES_MAP
 
 PROFILE_IMAGE_KEY_PREFIX = 'image_url'
 
 
+class LanguageProficiencySerializer(serializers.ModelSerializer):
+    """
+    Class that serializes the LanguageProficiency model for account
+    information.
+    """
+    display_name = serializers.SerializerMethodField("get_display_name")
+
+    class Meta:
+        model = LanguageProficiency
+        fields = ("code",)
+
+    def get_identity(self, data):
+        """
+        This is used in bulk updates to determine the identity of an object.
+        The default is to use the id of an object, but we want to override that
+        and consider the language code to be the canonical identity of a
+        LanguageProficiency model.
+        """
+        try:
+            return data.get('code', None)
+        except AttributeError:
+            return None
+
+    def validate_code(self, attrs, source):
+        """
+        Validate incoming language codes against the set of ISO 639 language
+        codes we support.
+        """
+        if source in attrs:
+            if attrs[source] not in settings.ALL_LANGUAGES_DICT:
```

Comment:
If you limit the model to these choices, you won't have to handle your own validator.